### PR TITLE
WIP: simplify vcpkg

### PR DESCRIPTION
**Pull request by bramtayl**
By bumping the vcpkg version and removing the custom portaudio port

On top of #1777. Only look at the last commit

### CODE REVIEW

File: InOut/CMakeLists.txt [hunk 1]
```diff
@@ -83,13 +83,12 @@ if(APPLE)
         target_link_libraries(rtauhal PRIVATE ${AUDIOUNIT_LIBRARY} ${COREFOUNDATION_LIBRARY} ${COREAUDIO_LIBRARY})
 endif()
 
-if(USE_ALSA AND LINUX)
-    if (USE_VCPKG)
-        find_package(ALSA MODULE REQUIRED)
-    else()
-        find_package(ALSA REQUIRED)
-    endif()
+if (NOT TARGET ALSA::ALSA)
```
Comment (bramtayl) [inline InOut/CMakeLists.txt:86]: `find_package(ALSA)` seems to only work once, and it might have already been called

File: InOut/CMakeLists.txt [hunk 2]
```diff
@@ -83,13 +83,12 @@ if(APPLE)
...
+    find_package(ALSA)
+endif()
+check_deps(USE_ALSA ALSA_FOUND)
```
Comment (bramtayl) [inline InOut/CMakeLists.txt:89]: Use check deps like other dependencies

File: vcpkg-configuration.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    "repository": "https://github.com/microsoft/vcpkg"
+  },
+  "registries": [
+    {
+      "kind": "artifact",
+      "location": "https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip",
+      "name": "microsoft"
+    }
+  ]
+}
```
Comment (bramtayl) [inline vcpkg-configuration.json:14]: This was auto-generated by `vcpkg/vcpkg new --application`

File: CMakeLists.txt [hunk 4]
```diff
@@ -5,31 +5,17 @@ cmake_policy(SET CMP0077 NEW)
...
-    elseif(APPLE)
-        set(VCPKG_TARGET_TRIPLET "x64-osx")
-    elseif (UNIX)
-        set(VCPKG_TARGET_TRIPLET "x64-linux-csound")
-    endif()
-
-    set(VCPKG_FEATURE_FLAGS versions)
-    set(X_VCPKG_APPLOCAL_DEPS_INSTALL ON)
-    set(VCPKG_OVERLAY_TRIPLETS ${CMAKE_CURRENT_SOURCE_DIR}/platform/vcpkg/triplets)
-    set(VCPKG_OVERLAY_PORTS ${CMAKE_CURRENT_SOURCE_DIR}/platform/vcpkg/ports)
```
Comment (stekyne) [inline CMakeLists.txt:25]: Why delete this?

File: InOut/CMakeLists.txt (hunk 1 above)
Comment (stekyne) [reply InOut/CMakeLists.txt:79]: Where is the code that finds Alsa?

File: InOut/CMakeLists.txt [hunk 5]
```diff
@@ -113,15 +105,15 @@ if(USE_PULSEAUDIO)
     make_plugin(rtpulse rtpulse.c ${PULSEAUDIO_LIBRARY} ${PULSESIMPLE_LIBRARY})
 endif()
 
-if(PORTAUDIO_FOUND OR USE_VCPKG)
+find_package(portaudio)
+if(USE_PORTAUDIO AND ((TARGET portaudio_static) OR (TARGET portaudio)))
```
Comment (stekyne) [inline InOut/CMakeLists.txt:109]: This syntax doesn't look great to me. I don't think this is the recommended way of doing things. Can you point to anywhere that says otherwise?

File: CMakeLists.txt (hunk 4 above)
Comment (bramtayl) [reply CMakeLists.txt:25]: Doesn't seem necessary?

File: InOut/CMakeLists.txt (hunk 1 above)
Comment (bramtayl) [reply InOut/CMakeLists.txt:81]: It's a build-in cmake module: https://cmake.org/cmake/help/latest/module/FindALSA.html

File: InOut/CMakeLists.txt (hunk 5 above)
Comment (bramtayl) [reply InOut/CMakeLists.txt:111]: Ref https://github.com/csound/csound/pull/1762#discussion_r1376936235

File: platform/vcpkg/ports/portaudio/portfile.cmake
Comment (stekyne) [inline platform/vcpkg/ports/portaudio/portfile.cmake:1]: Why did you delete this overlay port?

Comment (bramtayl) [reply platform/vcpkg/ports/portaudio/portfile.cmake:1]: I deleted it it looked like because the most recent version of vcpkg already included the patches from the custom port?
